File: Paradigmas de linguagem de programação/Go Lang/Retangulo/Retangulo.go
package main

import "fmt"

type Retangulo struct {
	comprimento, largura float64
}

func (r Retangulo) calcularArea() float64 {
	return r.comprimento * r.largura
}

func (r Retangulo) calcularPerimetro() float64 {
	return 2 * (r.comprimento + r.largura)
}

func main() {
	ret := Retangulo{comprimento: 200, largura: 300}
	fmt.Printf("Area: %.2f\n", ret.calcularArea())
	fmt.Printf("Perimetro: %.2f\n", ret.calcularPerimetro())
}

File: Paradigmas de linguagem de programação/Exercicios de sala de aula/Python/Retangulo/Retangulo.py
class Retangulo:

    def __init__(self, comprimento, largura):
        self.comprimento = comprimento
        self.largura = largura

    def calcular_area(self):
        return self.comprimento * self.largura

    def calcular_perimetro(self):
        return 2 * (self.comprimento + self.largura)

    def exibir_dados(self):
        print(f"O perímetro do retangulo é: {self.calcular_perimetro()}")
        print(f"A área do retangulo é: {self.calcular_area()}")

class main():
    ret = Retangulo(200, 300)
    ret.calcular_area()
    ret.calcular_perimetro()
    ret.exibir_dados()

if __name__ == "__main__":
    main()
